node_modules/col-assets/lib/api.js

```diff
@@ -1154,6 +1154,48 @@ var isDeletedOrHidden = module.exports.isDeletedOrHidden = function(asset) {
   return !asset.visible || !!asset.deleted_at || !(_.isEmpty(asset.categories) || _.find(asset.categories, 'visible'));
 };
 
+/**
+ * Returns true user is authorized to view the specified asset.
+ *
+ * @param  {User}         user              The user viewing the event
+ * @param  {Asset}        asset             The asset to check for hidden status
+ * @return {boolean}                        True if not deleted or not hidden from this particular user
+ */
+var canUserView = module.exports.canUserView = function(user, asset) {
+  var canUserView = false;
+
+  if (asset.deleted_at) {
+    canUserView = false;
+  } else if (user.is_admin || _.find(asset.users, {'id': user.id})) {
+    canUserView = true;
+  } else {
+    var hidden = !asset.visible || (!_.isEmpty(asset.categories) && !_.find(asset.categories, 'visible'));
```

Comment:
@johncrossman Since this compound Boolean is close to the one in `isDeletedOrHidden`, above, I wonder if we should move it into its own `isHidden` method and refer to that method in both places?

Reply:
@paulkerschen To preserve the (near) readability, I extracted only the `categories` logic. PR amended.